src/driver.ts

```diff
@@ -280,20 +285,19 @@ export class AppiumFlutterDriver extends BaseDriver<FlutterDriverConstraints> {
       flutterSystemPort:
         this.internalCaps.flutterSystemPort || (await getFreePort()),
     } as DriverCaps<FlutterDriverConstraints>;
-    const systemPort =
-      this.proxydriver instanceof XCUITestDriver &&
-      !this.proxydriver.isRealDevice()
-        ? null
-        : flutterCaps.flutterSystemPort!;
+    const systemPort = isIosSimulator ? null : flutterCaps.flutterSystemPort!;
```

Comment:
why the value is ignored for simulators? we could still use it same way it is done for WDA, e.g. allow the server to read it from env, so when flutter driver starts the app we just pass the desired port (default or flutterSystemPort value)

Reply:
then there will be no need to deal with files at all.